app.ts

```diff
@@ -0,0 +1,147 @@
+let recordCount: number, startIndex: number, endIndex: number;
+let columnHeaders: any[];
+let resizeTimer: number;
+
+window.onload = () => getRecordCount();
+ 
+//calculating how many records to display based on widow size
+function calRowPerView() {
```

Comment:
I would make this function set a value outside the function, something like `rowsPerView`. This will reduce the overhead of calling this function every time the value is required. This function is only ever required at least once, or when resizing the window.